boto/s3/inventory.py

```diff
@@ -316,6 +349,10 @@ def __init__(self, size=None, last_modified_date=None, storage_class=None,
         if intelligent_tiering_access_tier is not None:
             self.append("IntelligentTieringAccessTier")
 
+    def __eq__(self, other):
+        if other is None or not isinstance(other, InventoryOptionalFields): return False
+        return self.sort() == other.sort()

```

Comment:
```suggestion
        return sorted(self) == sorted(other)
```

Python 2.7's list `sort()` seems to sort the list in-place and return nothing so comparing `sort()` results will be always true even though two `InventoryOptionalFields` values have different ivars.

```python
$ python2

Python 2.7.16 (default, May  8 2021, 11:48:02) 
[GCC Apple LLVM 12.0.5 (clang-1205.0.19.59.6) [+internal-os, ptrauth-isa=deploy on darwin
Type "help", "copyright", "credits" or "license" for more information.

>>> from boto.s3.inventory import InventoryOptionalFields

>>> opt1 = InventoryOptionalFields(size=True)
>>> opt2 = InventoryOptionalFields(size=None)
>>> opt1 == opt2
True

>>> opt1.sort()
>>> opt2.sort()
>>> opt1.sort() == opt2.sort()
True
```

I found that a built-in function `sorted(list)` returns a sorted list, so I think we can use it instead.

```python
>>> sorted(opt1)
['Size']
>>> sorted(opt2)
[]
>>> sorted(opt1) == sorted(opt2)
False
```

Reply:
Thank for the review!
Fixed here. Please review it again.
https://github.com/cloudian/boto/pull/20/commits/5e0993a18649b5968021361163c52a949dfc2beb